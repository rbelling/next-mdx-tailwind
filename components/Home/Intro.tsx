export type IntroProps = {
  todaysDate: string;
};

const Intro = (props: IntroProps) => {
  const onClick = () => alert(`Clicked!`);
  return (
    <>
      <p className="text-italic">Today is {props.todaysDate}.</p>
      <br />
      <button
        className="rounded-full bg-orange-400	text-white px-4 py-2"
        onClick={onClick}
      >
        Click me
      </button>
    </>
  );
};

export default Intro;
