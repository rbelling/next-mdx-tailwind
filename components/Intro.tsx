'use client';
import MdxRenderer from '@/components/MdxRenderer';
import type { MDXRemoteSerializeResult } from 'next-mdx-remote';

const components = {
  Intro: () => <>I&apos;m a React component!</>,
};

const Intro = (props: MDXRemoteSerializeResult) => {
  return <MdxRenderer {...props} components={components} />;
};

export default Intro;
