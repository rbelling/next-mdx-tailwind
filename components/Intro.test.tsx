import {render} from '@testing-library/react';
import Intro from '@/components/Intro';

describe('Home', () => {
    it('matches snapshot', async () => {
        const props: Parameters<typeof Intro>[0] = {
            "compiledSource": "/*@jsxRuntime automatic @jsxImportSource react*/\nconst {jsx: _jsx, jsxs: _jsxs} = arguments[0];\nconst {useMDXComponents: _provideComponents} = arguments[0];\nfunction _createMdxContent(props) {\n const _components = Object.assign({\n p: \"p\",\n strong: \"strong\"\n }, _provideComponents(), props.components), {Intro} = _components;\n if (!Intro) _missingMdxReference(\"Intro\", true, \"1:37-1:46\");\n return _jsxs(_components.p, {\n children: [\"Some \", _jsx(_components.strong, {\n children: \"mdx\"\n }), \" text, with a component \", _jsx(Intro, {})]\n });\n}\nfunction MDXContent(props = {}) {\n const {wrapper: MDXLayout} = Object.assign({}, _provideComponents(), props.components);\n return MDXLayout ? _jsx(MDXLayout, Object.assign({}, props, {\n children: _jsx(_createMdxContent, props)\n })) : _createMdxContent(props);\n}\nreturn {\n default: MDXContent\n};\nfunction _missingMdxReference(id, component, place) {\n throw new Error(\"Expected \" + (component ? \"component\" : \"object\") + \" `\" + id + \"` to be defined: you likely forgot to import, pass, or provide it.\" + (place ? \"\\nItâ€™s referenced in your code at `\" + place + \"`\" : \"\"));\n}\n",
            "frontmatter": {},
            "scope": {}
        }
        const {container} = render(<Intro {...props} />);

        expect(container).toMatchSnapshot();
    });
});
