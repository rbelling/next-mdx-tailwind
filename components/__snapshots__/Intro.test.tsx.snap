// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home matches snapshot 1`] = `
<div>
  <div>
    <p>
      Some 
      <strong>
        mdx
      </strong>
       text, with a component 
      I'm a React component!
    </p>
  </div>
</div>
`;
